packages/opentelemetry-sdk-trace-base/src/utility.ts

```diff
@@ -61,20 +67,44 @@ export function mergeConfig(userConfig: TracerConfig): TracerConfig & {
 export function reconfigureLimits(userConfig: TracerConfig): TracerConfig {
   const spanLimits = Object.assign({}, userConfig.spanLimits);
 
+  const DEFAULT_CONFIG = loadDefaultConfig();
+
   /**
    * When span attribute count limit is not defined, but general attribute count limit is defined
    * Then, span attribute count limit will be same as general one
    */
-  if (spanLimits.attributeCountLimit == null && userConfig.generalLimits?.attributeCountLimit != null) {
-    spanLimits.attributeCountLimit = userConfig.generalLimits.attributeCountLimit;
+  if (
+    spanLimits.attributeCountLimit == null &&
+    (DEFAULT_CONFIG.spanLimits.attributeCountLimit === DEFAULT_ATTRIBUTE_COUNT_LIMIT) &&
+    (
+      userConfig.generalLimits?.attributeCountLimit != null ||
+      DEFAULT_CONFIG.generalLimits.attributeCountLimit !== DEFAULT_ATTRIBUTE_COUNT_LIMIT
+    )
+  ) {
+    const attributeCountLimit =
+     userConfig.generalLimits?.attributeCountLimit ??
+      DEFAULT_CONFIG.generalLimits.attributeCountLimit;
+
+    spanLimits.attributeCountLimit = attributeCountLimit;
   }
 
   /**
    * When span attribute value length limit is not defined, but general attribute value length limit is defined
    * Then, span attribute value length limit will be same as general one
    */
-  if (spanLimits.attributeValueLengthLimit == null && userConfig.generalLimits?.attributeValueLengthLimit != null) {
-    spanLimits.attributeValueLengthLimit = userConfig.generalLimits.attributeValueLengthLimit;
+  if (
+    spanLimits.attributeValueLengthLimit == null &&
+    (DEFAULT_CONFIG.spanLimits.attributeValueLengthLimit === DEFAULT_ATTRIBUTE_VALUE_LENGTH_LIMIT) &&
+    (
+      userConfig.generalLimits?.attributeValueLengthLimit != null ||
+      DEFAULT_CONFIG.generalLimits.attributeValueLengthLimit !== DEFAULT_ATTRIBUTE_VALUE_LENGTH_LIMIT
+    )
+  ) {
+    const attributeValueLengthLimit =
+      userConfig.generalLimits?.attributeValueLengthLimit ??
+        DEFAULT_CONFIG.generalLimits.attributeValueLengthLimit;
+
+    spanLimits.attributeValueLengthLimit = attributeValueLengthLimit;
```

Comment:
I'm not sure why we need to determine if the value is equal to `DEFAULT_ATTRIBUTE_VALUE_LENGTH_LIMIT` rather than simply nullish check. It will be way simpler with nullish check:

```
spanLimits.attributeValueLengthLimit = 
  userConfig.spanLimits?.attributeValueLengthLimit ?? 
  userConfig.generalLimits?.attributeValueLengthLimit ?? 
  env_config.spanLimits.attributeValueLengthLimit ?? 
  env_config.generalLimits.attributeValueLengthLimit;
```

Maybe we need to add a new `getEnv()` alternative that doesn't populate values if the value is not specified by the environs? 

Also, I'd find that if the user does specify the value identical to the DEFAULT value intentionally, the value is being ignored here anyways. I'd find this can be confusing.

Reply:
@legendecas It can't be a nullish check because `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT` and `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT` have a default value of `Infinity`. So as soon as `env_config.spanLimits.attributeValueLengthLimit ??` is checked,  it will use `Infinity` as the value for `spanLimits.attributeValueLengthLimit` even though the user set a custom value on `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT`.

Reply:
> Maybe we need to add a new getEnv() alternative that doesn't populate values if the value is not specified by the environs?

@legendecas Can you share more details on what you mean with this?

Reply:
@svetlanabrennan As you said, we can not perform a nullish check to determine the presence of an environ value because of the default value fallbacks. An alternative to this can be to use `parseEnvironment` from @opentelemetry/core directly to check the presence of an environment value safely with a nullish check.

Reply:
Awe that makes sense and is a much better approach. Thanks for the great recommendation.

Reply:
It's ready for review again.